### PR TITLE
[Composer] Fix: Fetch content of composer.json in specified working directory when calculating hash

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] fetches the content of `composer.json` in the specified working directory
* [x] invokes a `static` method `statically`
* [x] removes unused local variables
* [x] hardens a return type declaration

💁‍♂ Probably best reviewed commit by commit.

### CODE REVIEW

File: composer/helpers/src/Hasher.php [hunk 1]
```diff
@@ -5,17 +5,12 @@
...
+    public static function getContentHash(): string
     {
-        [$workingDirectory] = $args;
-
-        $io = new ExceptionIO();
-        $composer = Factory::create($io, $workingDirectory . '/composer.json');
-        $locker = $composer->getLocker();
-
-        return $locker->getContentHash(file_get_contents(Factory::getComposerFile()));
+        return Locker::getContentHash(file_get_contents(Factory::getComposerFile()));
```
Comment (localheinz) [inline composer/helpers/src/Hasher.php]: This method has been `static`, and still is, see

* https://github.com/composer/composer/blob/1.0.0/src/Composer/Package/Locker.php#L65-L102
* https://github.com/composer/composer/blob/1.9.0/src/Composer/Package/Locker.php#L65-L102

File: composer/helpers/src/Hasher.php [hunk 2]
```diff
@@ -5,17 +5,12 @@
...
 
 use Composer\Factory;
+use Composer\Package\Locker;
 
 class Hasher
 {
-    public static function getContentHash(array $args): ?string
+    public static function getContentHash(): string
     {
-        [$workingDirectory] = $args;
```
Comment (localheinz) [inline composer/helpers/src/Hasher.php]: If we need to be able to pass `$workingDirectory` to this method, then we need to make different changes.

Please let me know if that's the case.

File: composer/helpers/src/Hasher.php [hunk 3]
```diff
@@ -4,18 +4,16 @@
...
     {
         [$workingDirectory] = $args;
 
-        $io = new ExceptionIO();
-        $composer = Factory::create($io, $workingDirectory . '/composer.json');
-        $locker = $composer->getLocker();
+        $config = $workingDirectory . '/composer.json';
 
-        return $locker->getContentHash(file_get_contents(Factory::getComposerFile()));
+        return Locker::getContentHash(file_get_contents($config));
```
Comment (localheinz) [inline composer/helpers/src/Hasher.php]: This method has been `static`, and still is, see

* https://github.com/composer/composer/blob/1.0.0/src/Composer/Package/Locker.php#L65-L102
* https://github.com/composer/composer/blob/1.9.0/src/Composer/Package/Locker.php#L65-L102

**Comment by greysteil** [comment]
Nice! 👍

**Comment by localheinz** [comment]
Thank you, @greysteil!

**Pull request merged**